cmd/cloud/setup.go

```diff
@@ -308,3 +343,71 @@ func checkAPIKeys(astroClient astro.Client, coreClient astrocore.CoreClient, arg
 	}
 	return true, nil
 }
+
+func checkAPIToken(isDeploymentCmd bool, args []string) (bool, error) {
+	// check os variables
+	astroAPIToken := os.Getenv("ASTRO_API_TOKEN")
+	if astroAPIToken == "" {
+		return false, nil
+	}
+	if !isDeploymentCmd {
+		fmt.Println("Using an Astro API Token")
+	}
+
+	// get authConfig
+	c, err := context.GetCurrentContext() // get current context
+	if err != nil {
+		// set context
+		domain := defaultDomain
+		if !context.Exists(domain) {
+			err := context.SetContext(domain)
+			if err != nil {
+				return false, err
+			}
+		}
+
+		// Switch context
+		err = context.Switch(domain)
+		if err != nil {
+			return false, err
+		}
+
+		c, err = context.GetContext(domain) // get current context
+		if err != nil {
+			return false, err
+		}
+	}
+
+	err = c.SetContextKey("token", "Bearer "+astroAPIToken)
+	if err != nil {
+		return false, err
+	}
+
+	err = c.SetExpiresIn(time.Now().AddDate(1, 0, 0).Unix())
```

Comment:
Not sure if this is setting the right expiry time. Seeing this locally

` expiresin: 2077-05-26T06:38:54.144956-07:00`